File: Leetcode/2342. Max Sum of a Pair With Equal Sum of Digits.c
class Solution {
public:
    int getDigitSum(int n){
        int res=0;
        while(n>0){
            res+=n%10;
            n/=10;
        }
        return res;
    }
    int maximumSum(vector<int>& nums) {
        unordered_map<int,priority_queue<int>> m;
        for(int i:nums){
            m[getDigitSum(i)].push(i);
        }
        int res=-1;
        for(auto i:m){
            if(i.second.size()>=2){
                int p=i.second.top();
                // cout<<p<<"    ";
                i.second.pop();
                // cout<<i.second.top()<<"   "<<endl;
                p+=i.second.top();
                res=max(res,p);
            }
        }
        return res;
    }
};

File: Leetcode/2342. Max Sum of a Pair With Equal Sum of Digits.py
class Solution:
    def getDigitSum(self,num):
        res=0
        while num>0:
            res+=num%10
            num//=10
        return res
    def maximumSum(self, nums: List[int]) -> int:
        m={}
        for i in nums:
            t=self.getDigitSum(i)
            if t not in m:
                m[t]=[-i]
            else:
                m[t].append(-i)
        res=-1
        for h in m.values():
            if len(h)>=2:
                heapq.heapify(h)
                largest=heapq.nsmallest(2,h)
                res=max(-sum(largest),res)
        return res
        
        
